.github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml

```diff
@@ -0,0 +1,134 @@
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: PostCommit Python Nexmark Direct
+
+on:
+  issue_comment:
+    types: [created]
+  schedule:
+    - cron: '0 */6 * * *'
+  workflow_dispatch:
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.issue.number || github.event.pull_request.head.label || github.sha || github.head_ref || github.ref }}-${{ github.event.schedule || github.event.comment.body || github.event.sender.login}}'
+  cancel-in-progress: true
+
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: none
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+env:
+  GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_ACCESS_TOKEN }}
+  GRADLE_ENTERPRISE_CACHE_USERNAME: ${{ secrets.GE_CACHE_USERNAME }}
+  GRADLE_ENTERPRISE_CACHE_PASSWORD: ${{ secrets.GE_CACHE_PASSWORD }}
+  GRADLE_JAVA_COMMAND_ARGUMENTS: |
+    --manageResources=false
+    --monitorJobs=true
+    --bigQueryTable=nexmark
+    --bigQueryDataset=nexmark
+    --project=apache-beam-testing
+    --resourceNameMode=QUERY_RUNNER_AND_MODE
+    --exportSummaryToBigQuery=true
+    --tempLocation=gs://temp-storage-for-perf-tests/nexmark
+    --influxDatabase=beam_test_metrics
+    --influxHost=http://10.128.0.96:8086
+    --baseInfluxMeasurement=nexmark
+    --exportSummaryToInfluxDB=true
+    --influxRetentionPolicy=forever
+    --suite=SMOKE
+    --enforceEncodability=true
+    --enforceImmutability=true
+    --runner=DirectRunner
+    --numEvents=100000
+  GRADLE_PYTHON_COMMAND_ARGUMENTS: |
+    --monitor_jobs
+    --big_query_table=nexmark
+    --big_query_dataset=nexmark
+    --project=apache-beam-testing
+    --resource_name_mode=QUERY_RUNNER_AND_MODE
+    --export_summary_to_big_query
+    --temp_location=gs://temp-storage-for-perf-tests/nexmark
+    --influx_database=beam_test_metrics
+    --influx_host=http://10.128.0.96:8086
+    --base_influx_measurement=nexmark
+    --export_summary_to_influx_db
+    --influx_retention_policy=forever
+    --suite=SMOKE
+    --enforce_encodability
+    --enforce_immutability
+    --runner=DirectRunner
+    --num_events=100000
+
+jobs:
+  beam_PostCommit_Python_Nexmark_Direct:
+    name: ${{matrix.job_name}} (${{matrix.job_phrase}})
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    timeout-minutes: 240
+    strategy:
+      fail-fast: false
+      matrix:
+        job_name: [beam_PostCommit_Java_Nexmark_Direct]
+        job_phrase: [Run Python Direct Runner Nexmark Tests]
```

Comment:
I'm confused by the mixing of Java/Python in this job; why is one of these Java and one is Python?

Similarly, it seems like we're running both Java and Python benchmarks? Should those be in different workflow files?

Reply:
I didn't dig too deep into the intent, but based on the implementation and Jenkins runs I see that both tasks are being executed - https://ci-beam.apache.org/job/beam_PostCommit_Python_Nexmark_Direct/1003/consoleFull

Reply:
Ok, the job name should still be Python, though, right?

Reply:
Agree, it's confusing, but I see that Jenkins uses this exact comment phrase https://github.com/apache/beam/blob/master/.test-infra/jenkins/job_PostCommit_Python_Nexmark_Direct.groovy#L38.

Reply:
I'm talking about the job_name, not the job_phrase (`beam_PostCommit_Java_Nexmark_Direct` should be `beam_PostCommit_Python_Nexmark_Direct` I think)